plugins/woocommerce/src/Admin/Features/Settings/Transformer.php

```diff
@@ -0,0 +1,323 @@
+<?php
+/**
+ * WooCommerce Settings Data Transformer.
+ */
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Admin\Features\Settings;
+
+/**
+ * Transforms WooCommerce settings data into a structured format with logical groupings.
+ */
+class Transformer {
+	/**
+	 * Current group being processed.
+	 *
+	 * @var array|null
+	 */
+	private static ?array $current_group = null;
+
+	/**
+	 * Current checkbox group being processed.
+	 *
+	 * @var array|null
+	 */
+	private static ?array $current_checkbox_group = null;
+
+	/**
+	 * Transform settings data.
+	 *
+	 * @param array $raw_settings Raw settings data.
+	 *
+	 * @return array Transformed settings data.
+	 */
+	public static function transform( array $raw_settings ): array {
+		$transformed = array();
+
+		foreach ( $raw_settings as $tab_id => $tab ) {
+			// If the tab doesn't have sections, or the sections aren't an array, skip it.
+			if ( ! isset( $tab['sections'] ) || ! is_array( $tab['sections'] ) ) {
+				$transformed[ $tab_id ] = $tab;
+				continue;
+			}
+
+			$transformed[ $tab_id ] = array_merge(
+				$tab,
+				array( 'sections' => self::transform_sections( $tab['sections'] ) )
+			);
```

Comment:
I think we can optimize `array_merge` with the following change:
```suggestion
			$transformed[ $tab_id ] = $tab;
			$transformed[ $tab_id ]['sections'] = self::transform_sections( $tab['sections'] );
```

Reply:
Sounds good! Updated it 925e8d7f4355b3623207183583107ac146269eac. Thanks.